api/src/glfs-fops.c

```diff
@@ -843,18 +855,35 @@ pub_glfs_fstatat(struct glfs_fd *pglfd, const char *path, struct stat *stat,
         0,
     };
     int reval = 0;
+    int is_path_empty = 0;
 
     DECLARE_OLD_THIS;
     __GLFS_ENTRY_VALIDATE_FD(pglfd, invalid_fs);
 
+    is_path_empty = (flags & AT_EMPTY_PATH) == AT_EMPTY_PATH;
+
 retry:
     /* Retry case */
     if (subvol) {
         cleanup_fopat_args(pglfd, subvol, ret, &loc);
     }
 
-    subvol = setup_fopat_args(pglfd, path, !(flags & AT_SYMLINK_NOFOLLOW), &loc,
-                              &iatt, reval);
+    if (is_path_empty) {
```

Comment:
is the check be more like below? Or this is fine? Can you confirm from manpage? (would be similar for all fops)
```suggestion
    if (is_path_empty && !path) {
```

Reply:
yes it would be safer to add the check.
Added below,
`is_path_empty && path == '\0'`